File: sdk/nexastream.mjs
// NexaStream SDK (prioridade 1: JavaScript) — Item 26/58
export class NexaStream {
  constructor(o = {}) {
    this.core = o.core || 'http://localhost:3002';
    this.chain = o.chain || 'http://localhost:3008';
    this.explorer = o.explorer || 'http://localhost:3009';
    this.social = o.social || 'http://localhost:3011';
    this.mod = o.mod || 'http://localhost:3014';
    this.dao = o.dao || 'http://localhost:3015';
    this.token = o.token || null;
  }
  async _r(base, p, opts = {}) {
    const h = { ...(opts.headers || {}) };
    if (opts.json) h['Content-Type'] = 'application/json';
    if (this.token) h['Authorization'] = 'Bearer ' + this.token;
    const r = await fetch(base + p, { method: opts.method || 'GET', headers: h, body: opts.json ? JSON.stringify(opts.json) : opts.raw });
    const d = await r.json().catch(() => ({}));
    if (!r.ok) throw new Error(d.error || ('HTTP ' + r.status));
    return d;
  }
  // Auth
  async register(email, password, username) { const d = await this._r(this.core, '/api/auth/register', { method: 'POST', json: { email, password, username } }); this.token = d.token; return d; }
  async login(email, password) { const d = await this._r(this.core, '/api/auth/login', { method: 'POST', json: { email, password } }); this.token = d.token; return d; }
  // Vídeo
  videos() { return this._r(this.core, '/api/videos'); }
  video(id) { return this._r(this.core, '/api/videos/' + id); }
  search(q) { return this._r(this.core, '/api/search?q=' + encodeURIComponent(q)); }
  upload(fileBytes, title) { return this._r(this.core, '/api/videos/upload?title=' + encodeURIComponent(title), { method: 'PUT', raw: fileBytes }); }
  // Blockchain
  wallet() { return this._r(this.chain, '/api/chain/wallet', { method: 'POST', json: {} }); }
  balances() { return this._r(this.chain, '/api/chain/balances'); }
  verifyChain() { return this._r(this.chain, '/api/chain/verify'); }
  mine(miner) { return this._r(this.chain, '/api/chain/mine', { method: 'POST', json: { miner } }); }
  // Social / Moderação
  comment(videoId, username, content) { return this._r(this.social, '/api/social/comment', { method: 'POST', json: { videoId, username, content } }); }
  report(videoId, reason, reporter) { return this._r(this.mod, '/api/mod/report', { method: 'POST', json: { targetType: 'video', targetId: videoId, reason, reporter } }); }
  // DAO
  propose(p) { return this._r(this.dao, '/api/dao/proposal', { method: 'POST', json: p }); }
  vote(v) { return this._r(this.dao, '/api/dao/vote', { method: 'POST', json: v }); }
  proposals() { return this._r(this.dao, '/api/dao/proposals'); }
}
export default NexaStream;

File: sdk/python/nexastream.py
"""NexaStream Python SDK (Item 26, prioridade 2). Zero deps externas (so urllib)."""
import json
import urllib.request
import urllib.parse

class NexaStream:
    def __init__(self, core='http://localhost:3002', chain='http://localhost:3008',
                 explorer='http://localhost:3009', social='http://localhost:3011',
                 mod='http://localhost:3014', dao='http://localhost:3015',
                 nft='http://localhost:3016', kpi='http://localhost:3017'):
        self.u = dict(core=core, chain=chain, explorer=explorer, social=social,
                      mod=mod, dao=dao, nft=nft, kpi=kpi)
        self.token = None

    def _req(self, base, path, method='GET', body=None):
        data = json.dumps(body).encode() if body is not None else None
        r = urllib.request.Request(base + path, data=data, method=method)
        r.add_header('Content-Type', 'application/json')
        if self.token:
            r.add_header('Authorization', 'Bearer ' + self.token)
        with urllib.request.urlopen(r) as resp:
            return json.loads(resp.read().decode())

    # Auth
    def register(self, email, password, username):
        d = self._req(self.u['core'], '/api/auth/register', 'POST',
                      dict(email=email, password=password, username=username))
        self.token = d['token']; return d
    def login(self, email, password):
        d = self._req(self.u['core'], '/api/auth/login', 'POST',
                      dict(email=email, password=password))
        self.token = d['token']; return d

    # Video
    def videos(self): return self._req(self.u['core'], '/api/videos')
    def video(self, vid): return self._req(self.u['core'], '/api/videos/' + vid)
    def search(self, q): return self._req(self.u['core'], '/api/search?q=' + urllib.parse.quote(q))

    # Blockchain
    def wallet(self): return self._req(self.u['chain'], '/api/chain/wallet', 'POST', {})
    def balances(self): return self._req(self.u['chain'], '/api/chain/balances')
    def verify_chain(self): return self._req(self.u['chain'], '/api/chain/verify')

    # Ecossistema
    def comment(self, video_id, username, content):
        return self._req(self.u['social'], '/api/social/comment', 'POST',
                         dict(videoId=video_id, username=username, content=content))
    def report(self, video_id, reason, reporter):
        return self._req(self.u['mod'], '/api/mod/report', 'POST',
                         dict(targetType='video', targetId=video_id, reason=reason, reporter=reporter))
    def proposals(self): return self._req(self.u['dao'], '/api/dao/proposals')
    def nft_market(self): return self._req(self.u['nft'], '/api/nft/market')
    def kpi(self): return self._req(self.u['kpi'], '/api/kpi')
